ui/app/controllers/allocations/allocation/index.js

```diff
@@ -97,9 +97,9 @@ export default class IndexController extends Controller.extend(Sortable) {
   })
   stopAllocation;
 
-  @task(function* () {
+  @task(function* (allTasks) {
     try {
-      yield this.model.restart();
+      yield this.model.restart('', allTasks);
```

Comment:
`question` `thought`: You're passing in an empty string as a first argument here because we're looking to restart all tasks instead of only selecting one. That strikes me as a code smell for an API that should be refactored. But, I'm betting this is a backported change and we don't want to be risky like that.

Do you have any commentary about this so I can learn more?

Reply:
The API takes a body with either 

```json
{
  "Task": "my-task"
}
```

Or 

```json
{
  "AllTasks": true
}
```

If you try to set both it will result in an error. I didn't think too much about the changes to the function signatures but you're right, it doesn't look good with a random empty string 😅